maven-plugin/src/main/java/org/openmrs/maven/plugins/Fetch.java

```diff
@@ -0,0 +1,287 @@
+
+    private static final String DEFAULT_GROUP_ID = "org.openmrs.module";
+
+    public void executeTask() throws MojoExecutionException, MojoFailureException {
+        openmrsBintray = new OpenmrsBintray();
+        if (groupId == null) {
+            groupId = DEFAULT_GROUP_ID;
+        }
+        getData();
+        switch (projectType) {
+            case MODULE:
+                fetchMavenrepo(projectName, groupId, version, downloadDirectory);
+                break;
+            case OWA:
+                fetchBintray(REPOSITORY_OWNER, repository, projectName, version, downloadDirectory);
+                break;
+        }
+    }
+
+    private void getData() {
+        downloadDirectory = resolveDownloadDirectory(dir);
+
+        projectType = resolveProjectType(artifactId, owa);
+
+        if (projectType == ProjectType.OWA) {
+            repository = resolveRepositoryName(repository);
+        }
+
+        projectName = resolveProjectName(artifactId, owa);
+
+        if (version == null) {
+            version = resolveVersion(projectName);
+            if (version == null) {
+                switch (projectType) {
+                    case MODULE:
+                        version = askForMavenrepoProjectVersion(projectName, groupId);
+                        break;
+                    case OWA:
+                        version = askForBintrayProjectVersion(REPOSITORY_OWNER, repository, projectName);
+                        break;
+                }
+            }
+        }
+
+    }
+
+    private File resolveDownloadDirectory(String dirPath) {
+        //Resolve project directory if undefined
+        if (dirPath == null) {
+            dirPath = System.getProperty("user.dir");
+        }
+        return new File(dirPath);
+    }
+
+    private ProjectType resolveProjectType(String artifactId, String owa) {
+        //Resolve project type if defined
+        if (artifactId != null) {
+            return ProjectType.MODULE;
+        }
+        else if (owa != null) {
+            return ProjectType.OWA;
+        }
+
+        //Resolve project type if undefined
+        List<String> options = new ArrayList<>(Arrays.asList(
+                FETCH_MODULE_OPTION,
+                FETCH_OWA_OPTION
+        ));
+        if (projectType == null) {
+            String choice = wizard.promptForMissingValueWithOptions("What would you like to fetch?%s", null, "", options);
+            switch (choice) {
+                case FETCH_MODULE_OPTION:
+                    return ProjectType.MODULE;
+                case FETCH_OWA_OPTION:
+                    return ProjectType.OWA;
+            }
+        }
+        return null;
+    }
+
+    private String resolveProjectName(String artifactId, String owa) {
+        //Resolve project type and name if defined
+        if (artifactId != null) {
+            return artifactId;
+        }
+        else if (projectType == ProjectType.OWA) {
```

Comment:
not sure 100% about this but I think it would be cleaner to use ProjectType.OWA.equals(projecttype)